src/main/java/com/google/sps/data/TimeRange.java

```diff
@@ -168,6 +198,67 @@ public static int getTimeInMinutes(int hours, int minutes) {
     return (hours * 60) + minutes;
   }
 
+  public static int getTimeInMinutes(int day, int hours, int minutes){
+    if (day < 0 || day >= 7) {
+      throw new IllegalArgumentException("Day of the week must be between 0 and 6 (inclusive).");
+    }
+
+    if (hours < 0 || hours >= 24) {
+      throw new IllegalArgumentException("Hours can only be 0 through 23 (inclusive).");
+    }
+
+    if (minutes < 0 || minutes >= 60) {
+      throw new IllegalArgumentException("Minutes can only be 0 through 59 (inclusive).");
+    }
+
+    return (day * 24 * 60) + (hours * 60) + minutes;
+  }
+
+  /**
+   * Returns the day of the week in int form (0 being sunday, 6 being saturday)
+   * @param hours
+   * @param minutes
+   * @return int representing the day of the week
+   */
+  public static int getDayOfWeek(int hours, int minutes) {
+    if (hours < 0 || hours > 24 * 7) {
+      throw new IllegalArgumentException("Hours must be within a week period (between 0 and 167, inclusive).");
+    }
+
+    if (minutes < 0 || minutes >= 60) {
+      throw new IllegalArgumentException("Minutes can only be 0 through 59 (inclusive).");
+    }
+
+    return (int) Math.floorDiv(hours, 24);
+  }
+
+    /**
+   * Returns the day of the week in int form (0 being sunday, 6 being saturday)
+   * @param hours
+   * @param minutes
+   * @return int representing the day of the week
+   */
+  public static int getDayOfWeekHours(int hours) {
+    if (hours < 0 || hours > 24 * 7) {
+      throw new IllegalArgumentException("Time must be within a week period (between 0 and 167, inclusive).");
+    }
+
+    return (int) Math.floorDiv(hours, 24);
+  }
+
+  /**
+   * Returns the day of the week in int form (0 being sunday, 6 being saturday)
+   * @param minutes
+   * @return int representing the day of the week
+   */
+  public static int getDayOfWeekMinutes(int minutes) {
```

Comment:
This group of converter functions might be better named eg. `minutesToDayOfWeek`, `convertMinutesToWeekday` or something in that vein, to convey the fact that these functions take in some minute/hour/etc amount and convert it to a different unit. Typically `getXYZ` type names are reserved for getter methods that return some property of the object you're invoking it on.

Reply:
Resolved!